src/test/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/resource/ReportResourceTest.kt

```diff
@@ -1550,16 +1551,24 @@ class ReportResourceTest : SqsIntegrationTestBase() {
           assertNoDomainMessagesSent()
         }
 
-        @Test
-        fun `cannot add invalid object to a report`() {
-          webTestClient.post().uri(urlWithoutRelatedObjects)
-            .headers(setAuthorisation(roles = listOf("ROLE_MAINTAIN_INCIDENT_REPORTS"), scopes = listOf("write")))
-            .header("Content-Type", "application/json")
-            .bodyValue("{}")
-            .exchange()
-            .expectStatus().isBadRequest
-
-          assertNoDomainMessagesSent()
+        @DisplayName("cannot add invalid object to a report")
+        @TestFactory
+        fun `cannot add invalid object to a report`(): List<DynamicTest> {
+          val requests = invalidRequests.toMutableMap()
+          requests["empty request"] = "/related-objects/empty-object.json"
+          requests["invalid shape"] = "/related-objects/empty-array.json"
```

Comment:
these two could be inlined maybe as they're not specific of any of these resources, it may make the test a bit simpler.

Reply:
not cleanly cos the test tries to load the file at that path. it'd have to have a way to know whether it should 1) load a file or 2) use a string directly

Reply:
aren't the 2 "requests in this test just strings tho 🤔?

one [of the tests was using string literals](https://github.com/ministryofjustice/hmpps-incident-reporting-api/pull/124/files/7beec863dd26c39df358e91f1170b1f19c020042..5274daea0455abdcd8872441b3cefee6ff437ee6#diff-d6a0e69b7f1bf04510939e174746671e9585d2c1892aa614a0ca11211c1ac26dL1682) before no?
 
I don't mind if you think this is better or it's not possible to simplify, but I think there are a lot of fixtures files now ☺️

Reply:
actually i worked a way out. i don't think it makes things simpler really, but it does get rid of these stupid empty-ish json files